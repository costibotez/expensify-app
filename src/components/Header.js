import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" exact={true} activeClassName="is-active">Go home</NavLink>
        <NavLink to="/create" exact={true} activeClassName="is-active">Create expense</NavLink>
        <NavLink to="/edit" exact={true} activeClassName="is-active">Edit expense</NavLink>
    </header>
);

export default Header;