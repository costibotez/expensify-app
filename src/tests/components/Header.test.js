// react-test-renderer - Library to test React Components
import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';
// import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

test('should render header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find('h1').text()).toBe('Expensify');
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    // console.log(renderer.getRenderOutput());
});