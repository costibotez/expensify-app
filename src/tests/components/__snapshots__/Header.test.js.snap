// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Go home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/create"
  >
    Create expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/edit"
  >
    Edit expense
  </NavLink>
</header>
`;
